api/internal/gateway/admin/v1/handler/notification.go

```diff
@@ -1,22 +1,119 @@
 package handler
 
 import (
+	"fmt"
 	"net/http"
+	"strconv"
 
 	"github.com/and-period/furumaru/api/internal/gateway/admin/v1/request"
 	"github.com/and-period/furumaru/api/internal/gateway/admin/v1/response"
 	"github.com/and-period/furumaru/api/internal/gateway/admin/v1/service"
+	"github.com/and-period/furumaru/api/internal/gateway/util"
 	"github.com/and-period/furumaru/api/internal/messenger"
 	"github.com/and-period/furumaru/api/internal/messenger/entity"
+	"github.com/and-period/furumaru/api/internal/user"
 	"github.com/and-period/furumaru/api/pkg/jst"
 	"github.com/gin-gonic/gin"
 )
 
 func (h *handler) notificationRoutes(rg *gin.RouterGroup) {
 	arg := rg.Use(h.authentication())
+	arg.GET("", h.ListNotifications)
 	arg.POST("", h.CreateNotification)
 }
 
+func (h *handler) ListNotifications(ctx *gin.Context) {
+	const (
+		defaultLimit  = 20
+		defaultOffset = 0
+	)
+
+	limit, err := util.GetQueryInt64(ctx, "limit", defaultLimit)
+	if err != nil {
+		badRequest(ctx, err)
+		return
+	}
+	offset, err := util.GetQueryInt64(ctx, "offset", defaultOffset)
+	if err != nil {
+		badRequest(ctx, err)
+		return
+	}
+	orders, err := h.newNotificationOrders(ctx)
+	if err != nil {
+		badRequest(ctx, err)
+		return
+	}
+	since, err := util.GetQueryInt64(ctx, "since", 0)
+	if err != nil {
+		badRequest(ctx, err)
+		return
+	}
+	until, err := util.GetQueryInt64(ctx, "until", 0)
+	if err != nil {
+		badRequest(ctx, err)
+		return
+	}
+	onlyPublished, err := strconv.ParseBool(util.GetQuery(ctx, "OnlyPublished", "false"))
```

Comment:
あ、GetQueryBool 作ってなかったか、後で修正しときます🙏
ちなみに、query stringはローワーキャメルケースのほうがいいから ↓ のほうが適切かも

```suggestion
	onlyPublished, err := strconv.ParseBool(util.GetQuery(ctx, "onlyPublished", "false"))
```

Reply:
ありがとう！修正しました！